xds/internal/client/client_watchers_service.go

```diff
@@ -82,6 +83,10 @@ func (w *serviceUpdateWatcher) handleLDSResp(update ldsUpdate, err error) {
 		return
 	}
 
+	if w.rdsName == update.routeName {
+		return
```

Comment:
A small comment saying we are not cancelling and/or restarting the RDS watch if the LDS update is same.

Reply:
Done